File: src/main/java/com/alealogic/singleproxy/controller/DownloadController.java
package com.alealogic.singleproxy.controller;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

@RestController
public class DownloadController {

    @GetMapping(
            value = "alealogic-release",
            produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
    public byte[] getBinary(@Value("${binary.path}") String binaryPath) throws IOException {
        return Files.readAllBytes(Paths.get(binaryPath));
    }
}

File: src/main/java/com/alealogic/singleproxy/controller/DownloadController.kt
package com.alealogic.singleproxy.controller

import com.alealogic.singleproxy.service.FileService
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*
import java.io.IOException
import javax.servlet.http.HttpServletResponse

@RestController
@CrossOrigin(origins = ["http://localhost:3000", "http://10.0.2.2:3000"], allowCredentials = "true")
class DownloadController(private val fileService: FileService) {

    @GetMapping(value = ["alealogic-release"], produces = [MediaType.APPLICATION_OCTET_STREAM_VALUE])
    @Throws(IOException::class)
    fun getBinary(
        @CookieValue("token") sessionToken: String,
        response: HttpServletResponse
    ): ByteArray = fileService.getBinaryBySessionToken(sessionToken).let {
        response.setHeader("Content-Disposition", "attachment; filename=" + it.first)
        it.second
    }

    @GetMapping("install")
    fun getInstallationScript(@RequestParam("id") sessionToken: String): String =
        fileService.getInstallationScript(sessionToken)

}